ktor-server/ktor-server-plugins/ktor-server-thymeleaf/jvm/test/io/ktor/server/thymeleaf/ThymeleafTest.kt

```diff
@@ -17,34 +19,36 @@ import java.util.*
 import java.util.zip.*
 import kotlin.test.*
 
-@Suppress("DEPRECATION")
 class ThymeleafTest {
     @Test
     fun testName() {
-        withTestApplication {
-            application.setUpThymeleafStringTemplate()
-            application.install(ConditionalHeaders)
-            application.routing {
-                val model = mapOf("id" to 1, "title" to "Hello, World!")
+        testApplication {
+            application {
+                setUpThymeleafStringTemplate()
+                install(ConditionalHeaders)
+                routing {
+                    val model = mapOf("id" to 1, "title" to "Hello, World!")
 
-                get("/") {
-                    call.respond(ThymeleafContent(STRING_TEMPLATE, model, "e"))
+                    get("/") {
+                        call.respond(ThymeleafContent(STRING_TEMPLATE, model, "e"))
+                    }
                 }
             }
 
-            handleRequest(HttpMethod.Get, "/").response.let { response ->
-                assertNotNull(response.content)
-                val expectedContent = listOf("<p>Hello, 1</p>", "<h1>Hello, World!</h1>")
-                assertEquals(expectedContent, response.content!!.lines())
+            val response = client.get("/")
+            val content = response.bodyAsText()
+            assertNotNull(content)
+            val expectedContent = listOf("<p>Hello, 1</p>", "<h1>Hello, World!</h1>")
+            assertEquals(expectedContent, content.lines())
 
-                val contentTypeText = assertNotNull(response.headers[HttpHeaders.ContentType])
-                assertEquals(ContentType.Text.Html.withCharset(Charsets.UTF_8), ContentType.parse(contentTypeText))
-                assertEquals("\"e\"", response.headers[HttpHeaders.ETag])
-            }
+            val contentTypeText = assertNotNull(response.headers[HttpHeaders.ContentType])
+            assertEquals(ContentType.Text.Html.withCharset(Charsets.UTF_8), ContentType.parse(contentTypeText))
+            assertEquals("\"e\"", response.headers[HttpHeaders.ETag])
         }
     }
 
     @Test
+    @Suppress("DEPRECATION")
     fun testCompression() {
```

Comment:
any reason why this test is not rewritten to new API?

Reply:
Sorry, I've postponed this test and forgot about it. It's rewritten now.